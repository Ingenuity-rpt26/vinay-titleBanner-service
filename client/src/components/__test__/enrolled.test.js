import React from 'react';
import ReactDOM from 'react-dom';
import Enrolled from '../enrolled';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Enrolled></Enrolled>, div);
});


it('render enrolled correctly', () => {
  const {getByTestId} = render(<Enrolled enrolled= {5}></Enrolled>);
  expect(getByTestId('enrolled')).toHaveTextContent(5);
});

//test DOM cleanup
it('render enrolled correctly', () => {
  const {getByTestId} = render(<Enrolled enrolled= {10}></Enrolled>);
  expect(getByTestId('enrolled')).toHaveTextContent(10);
});

//test snapshot
it('matches snapshot', () => {
  const tree = renderer.create(<Enrolled enrolled={100}></Enrolled>).toJSON();
  expect(tree).toMatchSnapshot();
});

