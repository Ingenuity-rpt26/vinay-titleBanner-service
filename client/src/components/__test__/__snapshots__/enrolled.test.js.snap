// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="enrolled"
>
  <div>
    100
  </div>
</div>
`;
